internal/storage/storage_handle_test.go

```diff
@@ -121,10 +122,33 @@ func (t *StorageHandleTest) TestNewStorageHandleWithCustomEndpoint() {
 	t.invokeAndVerifyStorageHandle(sc)
 }
 
-// This will fail while fetching the token-source, since key-file doesn't exist.
 func (t *StorageHandleTest) TestNewStorageHandleWhenCustomEndpointIsNil() {
 	sc := storageutil.GetDefaultStorageClientConfig()
 	sc.CustomEndpoint = nil
+	sc.DisableAuth = true
```

Comment:
Because the name TestNewStorageHandleWhenCustomEndpointIsNil doesn't specify if we are changing anything other than CustomEndpoint, so we should use default value of other parameters. So DisableAuth shouldn't be set here.

TestNewStorageHandleWhenCustomEndpointIsNilAndDisableAuthIsFalse should be changed to TestNewStorageHandleWhenCustomEndpointIsNilAndDisableAuthIsTrue